doc/RIF_counters.md

```diff
@@ -0,0 +1,261 @@
+# Router interface counters in SONiC
+### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About this Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definitions/Abbreviation](#definitionsabbreviation)
+
+  * [1 Requirements](#1-overview)
+
+  * [2 CLI format](#2-cli-format)
+
+  * [3 Modules Design](#3-modules-design)
+    * [3.1 Modules that need to be updated](#31-modules-that-need-to-be-updated)
+      * [3.1.1 Counter DB](#311-counter-db)
+      * [3.1.2 CLI](#312-cli)
+      * [3.1.3 Orchestration Agent](#313-orchestration-agent)
+      * [3.1.4 Flex Counter](#314-flex-counter)
+      * [3.1.5 SAI](#318-sai)
+
+  * [4 Open questions](#4-open-questions)
+
+# List of Tables
+* [Table 1: Revision](#revision)
+* [Table 2: Abbreviations](#definitionsabbreviation)
+
+###### Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 |             | Mykola Faryma      | Initial version                   |
+
+# About this Manual
+This document provides general information about RIF counter feature in SONiC.
+# Scope
+This document describes the high level design of the RIF counters.
+# Definitions/Abbreviation
+###### Table 2: Abbreviations
+| Definitions/Abbreviation | Description                                |
+|--------------------------|--------------------------------------------|
+| RIF                      | Router interface                           |
+| FC                       | Flex counter                               |
+| API                      | Application Programmable Interface         |
+| SAI                      | Switch Abstraction Interface               |
+
+# 1 Requirements
+
+Add new CLI commands to show L3 interfaces counters. L3 interface counters refers to router port as well as vlan interface.
+The information should be: RX: OK, ERR in packets and bytes. TX: OK, ERR in packets and bytes 
+ 
+# 2 CLI format
+The CLI command format and example output
+
+```
+$ show interfaces counters rif [OPTOINS] [interface_name]
+
+ Show interface counters
+
+Options:
+
+  -p, --period TEXT
+
+  --verbose          Enable verbose output
+
+  -?, -h, --help     Show this message and exit.
+```
+
+Note: the 'show interfaces counters' has '-c, --clear' option. From the CLI point of view, this should be available from 'sonic-clear' CLI. (There already is 'sonic-clear interfaces counters')
```

Comment:
As we discussed in meeting, this would need to be retained as it is being used in production scripts